lib/filewatch/watched_file.rb

```diff
@@ -0,0 +1,180 @@
+require "filewatch/buftok"
+
+module FileWatch
+  class WatchedFile
```

Comment:
Naming-wise I'm not sure if `WatchedFile` is a good choice since the it may or may not be watched at any point in time.

Since this is scoped to FileWatch and read operations are called in these objects. what about naming this "FileWatch::File" ?